// @farcaster/frame-core/dist/schemas/manifest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.domainManifestSchema = exports.domainFrameConfigSchema = void 0;
const zod_1 = require("zod");
const shared_1 = require("./shared");
exports.domainFrameConfigSchema = zod_1.z.object({
    // 0.0.0 and 0.0.1 are not technically part of the spec but kept for
    // backwards compatibilty. next should always resolve to the most recent
    // schema version.
    version: zod_1.z.union([
        zod_1.z.literal('0.0.0'),
        zod_1.z.literal('0.0.1'),
        zod_1.z.literal('1'),
        zod_1.z.literal('next'),
    ]),
    name: shared_1.frameNameSchema,
    iconUrl: shared_1.secureUrlSchema,
    homeUrl: shared_1.secureUrlSchema,
    imageUrl: shared_1.secureUrlSchema.optional(),
    buttonTitle: shared_1.buttonTitleSchema.optional(),
    splashImageUrl: shared_1.secureUrlSchema.optional(),
    splashBackgroundColor: shared_1.hexColorSchema.optional(),
    webhookUrl: shared_1.secureUrlSchema.optional(),
});
exports.domainManifestSchema = zod_1.z.object({
    accountAssociation: shared_1.encodedJsonFarcasterSignatureSchema,
    frame: exports.domainFrameConfigSchema.optional(),
});
